### PR TITLE
Careful release of SchemaRulesArgs

### PR DESCRIPTION
**Pull request by rafie**

### CODE REVIEW

File: src/rules.c [hunk 1]
```diff
@@ -53,6 +53,22 @@ void SchemaRuleArgs_Free(SchemaRuleArgs *rule_args) {
   rm_free(rule_args);
 }
 
+void SchemaRulesArgs_Free(RedisModuleCtx *ctx) {
```
Comment (ashtul) [inline src/rules.c:56]: You can use `RSDummyContext`

Comment (rafie) [reply src/rules.c:56]: I wanted to use the context from one of the original calls. It gives better logs.

Comment (rafie) [reply src/rules.c:56]: But we can still use dummy instead of NULL, right?

File: src/spec.c [hunk 2]
```diff
@@ -2017,6 +2003,7 @@ static void onFlush(RedisModuleCtx *ctx, RedisModuleEvent eid, uint64_t subevent
   }
   IndexSpec_CleanAll();
   Dictionary_Clear();
+  SchemaRulesArgs_Free(ctx);
```
Comment (MeirShpilraien) [inline src/spec.c:2006]: Maybe call it `LegacySchemaRulesArgs_Free`?  Also I am not sure why you need it here, maybe we are hiding a problem?

File: src/module.c [hunk 3]
```diff
@@ -1060,10 +1060,7 @@ void __attribute__((destructor)) RediSearch_CleanupModule(void) {
     dictRelease(legacySpecDict);
     legacySpecDict = NULL;
   }
-  if (legacySpecRules) {
-    dictRelease(legacySpecRules);
-    legacySpecRules = NULL;
-  }
+  SchemaRulesArgs_Free(NULL);
```
Comment (MeirShpilraien) [inline src/module.c:1063]: Here also I am not sure why you need it?

File: src/spec.c (hunk 2 above)
Comment (rafie) [reply src/spec.c:2006]: Will rename and remove from `onFlash` and we'll see what happens.

**Review by ashtul — APPROVED** [review]

**Pull request merged**